src/librustc_typeck/check/coercion.rs

```diff
@@ -828,6 +828,15 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
         self.probe(|_| coerce.coerce(source, target)).is_ok()
     }
 
+    pub fn deref_steps(&self, expr_ty: Ty<'tcx>, target: Ty<'tcx>) -> Option<usize> {
```

Comment:
Please leave some documentation on this function. I think it would help future callers to know in what cases this returns `None`

Reply:
Done!